src/main/java/de/tum/in/www1/artemis/service/StudentScoreService.java

```diff
@@ -0,0 +1,127 @@
+import de.tum.in.www1.artemis.domain.User;
+import de.tum.in.www1.artemis.domain.participation.StudentParticipation;
+import de.tum.in.www1.artemis.domain.scores.StudentScore;
+import de.tum.in.www1.artemis.repository.ExerciseRepository;
+import de.tum.in.www1.artemis.repository.StudentScoreRepository;
+
+@Service
+public class StudentScoreService {
+
+    private final Logger log = LoggerFactory.getLogger(StudentScoreService.class);
+
+    private final StudentScoreRepository studentScoreRepository;
+
+    private final ExerciseRepository exerciseRepository;
+
+    public StudentScoreService(StudentScoreRepository studentScoreRepository, ExerciseRepository exerciseRepository) {
+        this.studentScoreRepository = studentScoreRepository;
+        this.exerciseRepository = exerciseRepository;
+    }
+
+    /**
+     * Returns all StudentScores for exercise.
+     *
+     * @param exercise the exercise
+     * @return list of student score objects for that exercise
+     */
+    public List<StudentScore> getStudentScoresForExercise(Exercise exercise) {
+        return studentScoreRepository.findAllByExercise(exercise);
+    }
+
+    /**
+     * Returns all StudentScores for course.
+     *
+     * @param course course
+     * @return list of student score objects for that course
+     */
+    public List<StudentScore> getStudentScoresForCourse(Course course) {
+        return studentScoreRepository.findAllByExerciseIdIn(course.getExercises());
+    }
+
+    /**
+     * Delete all StudentScores for exercise.
+     *
+     * @param exercise exercise
+     */
+    public void deleteStudentScoresForExercise(Exercise exercise) {
+        var scores = getStudentScoresForExercise(exercise);
+
+        for (StudentScore score : scores) {
+            studentScoreRepository.delete(score);
+        }
+    }
+
+    /**
+     * Returns one StudentScores for exercise and student if one exists.
+     *
+     * @param student student user
+     * @param exercise exercise
+     * @return student score for student and exercise if it exists
+     */
+    public Optional<StudentScore> getStudentScoreForStudentAndExercise(User student, Exercise exercise) {
+        return studentScoreRepository.findByStudentAndExercise(student, exercise);
+    }
+
+    /**
+     * Removes StudentScore for result deletedResult.
+     *
+     * @param deletedResult result to be deleted
+     */
+    public void removeResult(Result deletedResult) {
+        deletedResult.setStudentScore(null);
+        studentScoreRepository.deleteByResult(deletedResult);
+    }
+
+    /**
+     * Updates all StudentScores for result updatedResult.
+     *
+     * @param updatedResult result to be updated
+     */
+    public void updateResult(Result updatedResult) {
+        // ignore results without score or participation
+        if (updatedResult.getScore() == null || updatedResult.getParticipation() == null || !Boolean.TRUE.equals(updatedResult.isRated())) {
```

Comment:
Just as a side note but I don't really understand why we use `!Boolean.TRUE.equals`. Wouldn't `Boolean.FALSE.equals` we easier to read?